config.go

```diff
@@ -33,6 +33,17 @@ type Config struct {
 			Config *tls.Config
 		}
 
+		// SASL based authentication with broker. While there are multiple SASL authentication methods
+		// the current implementation is limited to plaintext (SASL/PLAIN) authentication
```

Comment:
since we currently only support plain-text, should we enforce that SASL is only used over HTTPS to avoid leaking credentials on the wire?